pkg/build/builder/cmd/builder.go

```diff
@@ -45,9 +45,13 @@ func run(builderFactory factoryFunc) {
 			glog.Fatalf("Cannot parse build URL: %s", build.Spec.Source.Git.URI)
 		}
 		scmAuths := auths(sourceURL)
-		if err := setupSourceSecret(build.Spec.Source.SourceSecret.Name, scmAuths); err != nil {
+		url, err := setupSourceSecret(build.Spec.Source.SourceSecret.Name, scmAuths)
+		if err != nil {
 			glog.Fatalf("Cannot setup secret file for accessing private repository: %v", err)
 		}
+		if url != nil {
+			build.Spec.Source.Git.URI = url.String()
```

Comment:
is stomping the entire URL correct? we need good tests to make sure all the non-auth-related bits of the URL are preserved (path, etc)